internal/core/vulndb/scan/scan_controller.go

```diff
@@ -192,4 +187,48 @@ func (s *httpController) Scan(c core.Context) error {
 		AmountClosed: amountClose,
 		Flaws:        utils.Map(newState, flaw.FlawToDto),
 	})
+
+}
+
+func (s *httpController) Scan(c core.Context) error {
+	bom := new(cdx.BOM)
+	decoder := cdx.NewBOMDecoder(c.Request().Body, cdx.BOMFileFormatJSON)
+	if err := decoder.Decode(bom); err != nil {
+		return err
+	}
+	if err := decoder.Decode(bom); err != nil {
+		return err
+	}
+
+	err := ScanHelpFunction(c, normalize.FromCdxBom(bom, true), s)
+
+	return c.JSON(200, err)
+
+}
+
+func (s *httpController) ManualSbomScan(c core.Context) error {
+
+	var maxSize int = 16 * 1024 * 1024 //Max Upload Size 16mb
```

Comment:
```suggestion
	var maxSize int64 = 16 * 1024 * 1024 //Max Upload Size 16mb
```

Thsi way you avoid the casting in line 212